src/elements/boxandwhiskers.js

```diff
@@ -75,56 +75,70 @@ const BoxAndWiskers = Chart.elements.BoxAndWhiskers = ArrayElementBase.extend({
   },
   _drawBoxPlot(vm, boxplot, ctx, vert) {
     if (vert) {
-      const x = vm.x;
-      const width = vm.width;
-      const x0 = x - width / 2;
-      if (boxplot.q3 > boxplot.q1) {
-        ctx.fillRect(x0, boxplot.q1, width, boxplot.q3 - boxplot.q1);
-        ctx.strokeRect(x0, boxplot.q1, width, boxplot.q3 - boxplot.q1);
-      } else {
-        ctx.fillRect(x0, boxplot.q3, width, boxplot.q1 - boxplot.q3);
-        ctx.strokeRect(x0, boxplot.q3, width, boxplot.q1 - boxplot.q3);
-      }
-      ctx.beginPath();
-      ctx.moveTo(x0, boxplot.whiskerMin);
-      ctx.lineTo(x0 + width, boxplot.whiskerMin);
-      ctx.moveTo(x, boxplot.whiskerMin);
-      ctx.lineTo(x, boxplot.q1);
-      ctx.moveTo(x0, boxplot.whiskerMax);
-      ctx.lineTo(x0 + width, boxplot.whiskerMax);
-      ctx.moveTo(x, boxplot.whiskerMax);
-      ctx.lineTo(x, boxplot.q3);
-      ctx.moveTo(x0, boxplot.median);
-      ctx.lineTo(x0 + width, boxplot.median);
-      ctx.closePath();
-      ctx.stroke();
+      this._drawBoxPlotVert(vm, boxplot, ctx);
     } else {
-      const y = vm.y;
-      const height = vm.height;
-      const y0 = y - height / 2;
-      if (boxplot.q3 > boxplot.q1) {
-        ctx.fillRect(boxplot.q1, y0, boxplot.q3 - boxplot.q1, height);
-        ctx.strokeRect(boxplot.q1, y0, boxplot.q3 - boxplot.q1, height);
-      } else {
-        ctx.fillRect(boxplot.q3, y0, boxplot.q1 - boxplot.q3, height);
-        ctx.strokeRect(boxplot.q3, y0, boxplot.q1 - boxplot.q3, height);
-      }
-      ctx.beginPath();
-      ctx.moveTo(boxplot.whiskerMin, y0);
-      ctx.lineTo(boxplot.whiskerMin, y0 + height);
-      ctx.moveTo(boxplot.whiskerMin, y);
-      ctx.lineTo(boxplot.q1, y);
-      ctx.moveTo(boxplot.whiskerMax, y0);
-      ctx.lineTo(boxplot.whiskerMax, y0 + height);
-      ctx.moveTo(boxplot.whiskerMax, y);
-      ctx.lineTo(boxplot.q3, y);
-      ctx.moveTo(boxplot.median, y0);
-      ctx.lineTo(boxplot.median, y0 + height);
-      ctx.closePath();
-      ctx.stroke();
+      this._drawBoxPlotHoriz(vm, boxplot, ctx);
     }
 
   },
+  _drawBoxPlotVert(vm, boxplot, ctx) {
+    const x = vm.x;
+    const width = vm.width;
+    const x0 = x - width / 2;
+    if (boxplot.q3 > boxplot.q1) {
+      ctx.fillRect(x0, boxplot.q1, width, boxplot.q3 - boxplot.q1);
+      ctx.strokeRect(x0, boxplot.q1, width, boxplot.q3 - boxplot.q1);
+    } else {
+      ctx.fillRect(x0, boxplot.q3, width, boxplot.q1 - boxplot.q3);
+      ctx.strokeRect(x0, boxplot.q3, width, boxplot.q1 - boxplot.q3);
+    }
+    ctx.beginPath();
+    ctx.moveTo(x0, boxplot.whiskerMin);
+    ctx.lineTo(x0 + width, boxplot.whiskerMin);
+    ctx.moveTo(x, boxplot.whiskerMin);
+    ctx.lineTo(x, boxplot.q1);
+    ctx.moveTo(x0, boxplot.whiskerMax);
+    ctx.lineTo(x0 + width, boxplot.whiskerMax);
+    ctx.moveTo(x, boxplot.whiskerMax);
+    ctx.lineTo(x, boxplot.q3);
+    ctx.closePath();
+    ctx.stroke();
+    ctx.strokeStyle = vm.medianColor;
```

Comment:
please restore the previous set `ctx.strokeStyle` after rendering the median line. or use `ctx.save()` and `ctx.restore()` to ensure no side effect for follow up methods, like a changed `strokeStyle`

Reply:
Context is now saved, as well as the median line not being the last to be drawn, so subsequent calls are no longer affected